autoload/go/test_test.vim

```diff
@@ -0,0 +1,109 @@
+        \ {'lnum': 16, 'bufnr': 2, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'helper badness'},
+        \ {'lnum': 20, 'bufnr': 2, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'this is an error'},
+        \ {'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'and a second line, too'},
+        \ {'lnum': 25, 'bufnr': 2, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'this is a sub-test error'},
+        \ {'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'and a second line, too'},
+        \ {'lnum': 40, 'bufnr': 2, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'panic: concurrent fail'}
+      \ ]
+
+  call s:test(expected, 0)
+endfunc
+
+func! Test_GoTestVerbose() abort
+  let expected = [
+        \ {'lnum': 12, 'bufnr': 2, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'log message'},
+        \ {'lnum': 13, 'bufnr': 2, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'sub badness'},
+        \ {'lnum': 15, 'bufnr': 2, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'badness'},
+        \ {'lnum': 16, 'bufnr': 2, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'helper badness'},
+        \ {'lnum': 20, 'bufnr': 2, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'this is an error'},
+        \ {'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'and a second line, too'},
+        \ {'lnum': 25, 'bufnr': 2, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'this is a sub-test error'},
+        \ {'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'and a second line, too'},
+        \ {'lnum': 31, 'bufnr': 2, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'goodness'},
+        \ {'lnum': 40, 'bufnr': 2, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'panic: concurrent fail'}
+      \ ]
+
+  call s:test(expected, 1)
+endfunc
+
+func! s:test(expected, verbose) abort
+  if has('nvim')
+    " don't test neovim; it seems to mostly work, but the errors it displays
+    " are slightly out of order due to the way it interleaves stderr and
+    " stdout lines from async jobs, and it doesn't implement support for
+    " 'errorformat' correctly; multi-line errors get handled differently than
+    " they do for Vim, and the messages produce by neovim include characters
+    " that get trimmed out of Vim's messages.
+    return
+  endif
+
+  let $GOPATH = fnameescape(expand("%:p:h")) . '/test-fixtures/test'
+  silent exe 'e ' . $GOPATH . '/src/play/play_test.go'
```

Comment:
There is the `gotest#load_fixture()` helper which you should be able to use for this:

```
try
  let l:tmp = gotest#load_fixture('test/src/play/play_test.go')
finally
  call delete(l:tmp, 'rf')
endtry
```

This will copy the fixture to a tmp dir, and sets GOPATH. This means it will never write to the vim-go source directory, which is better when running tests, I think.

Reply:
loadfixture only copies one file, but this test depends on more files than just the one being loaded.

Reply:
Oh, right. Should probably add support for multiple paths in that case. I'll look at that later (this is fine for now).